docs/source/getting_started.rst

```diff
@@ -320,35 +320,140 @@ syntax like so:
-and checking that the argument to :py:func:`chr` has a valid type.
+    audited = AuditedBankAccount('Charlie', 300)
+    transfer(account_1, audited, 100)   # Type checks!
 
-Mypy complains if it can't find a stub (or a real module) for a
-library module that you import. Some modules ship with stubs or inline
-annotations that mypy can automatically find, or you can install
-additional stubs using pip (see :ref:`fix-missing-imports` and
-:ref:`installed-packages` for the details). For example, you can install
-the stubs for the ``requests`` package like this:
+This behavior is actually a fundamental aspect of the PEP 484 type system: when 
+we annotate some variable with a type ``T``, we are actually telling mypy that
+variable can be assigned an instance of ``T``, or an instance of a *subclass* of ``T``.
+The same rule applies to type hints on parameters or fields.
+
+See :ref:`class-basics` to learn more about how to work with code involving classes.
+
+
+.. _stubs-intro:
+
+Stubs files and typeshed
+************************
+
+Mypy also understands how to work with classes found in the standard library.
+For example, here is a function which uses the ``Path`` object from the
+`pathlib standard library module <https://docs.python.org/3/library/pathlib.html>`_:
+
+.. code-block:: python
+
+    from pathlib import Path
+
+    def load_template(template_path: Path, name: str) -> str:
+        # Mypy understands that 'file_path.read_text()' returns a str...
+        template = template_path.read_text()
+
+        # ...so understands this line type checks.
+        return template.replace('USERNAME', name)
+
+This behavior may surprise you if you're familiar with how
+Python internally works. The standard library does not use type hints
+anywhere, so how did mypy know that ``Path.read_text()`` returns a ``str``,
+or that ``str.replace(...)`` accepts exactly two ``str`` arguments?
+
+The answer is that mypy comes bundled with *stub files* from the
+the `typeshed <https://github.com/python/typeshed>`_ project.
+
+A *stub file* is a file containing a skeleton of the public interface
+of that Python module, including classes, variables, functions -- and
+most importantly, their types. Typeshed is a collection of stub files
+for modules in the standard library and select third party libraries
+and is what mypy used to identify the correct types for our example above.
+
+Mypy can understand third party libraries in the same way as long as those
+libraries come bundled with stub files or have inline annotations that mypy
+can automatically find. (see :ref:`fix-missing-imports` and
+:ref:`installed-packages` for the details). However, if the third party library
+does *not* come bundled with type hints, mypy will not try and guess what
+the types are: it'll assume the entire library is
+:ref:`dynamically typed <dynamic-typing>` and report an error
+whenever you import the library.
+
```

Comment:
@JukkaL I tried to merge the conflicts. But I am not sure about this parts.

Reply:
I think this section (from lines 452-487) should be moved to `running_mypy.rst`, or maybe just cut altogether. With the changes made in this PR, it's a very different topic to the rest of the article, and I don't think we need it here since it's covered in depth elsewhere.

Reply:
I removed the changes by Michael. But I keep the current example since it's quite useful for beginners IMHO.

Reply:
I'd vote for cutting this section entirely. At this point, typeshed is good enough that users don't need to know at all about typeshed and beginner users don't really need to know about stubs to get started